### PR TITLE
[DNM] expiremental branch with verbose logging

### PR DESCRIPTION
**Pull request by riastrad**
This PR is not meant to be merged to master. It has experimental logging to help better understand behavior in areas where `@mapbox/locking` is being used. The goal will be to release a dev. version of this branch, test where useful/relevant, and then close this PR. 

cc/ @mapbox/static-apis

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -37,8 +37,8 @@ function createStaleCacher(cache, locks, cacheStats, loader) {
     return function(id, callback) {
         // Stringify objects.
         var key = JSON.stringify(id);
-
-        // Instance is in LRU cache.
+        (!loader.name) ? lockedFunction = 'tlMapbox' : lockedFunction = loader.name;
```
Comment (springmeyer) [inline index.js]: Does this need a `let lockedFunction;` at the top @riastrad ?

Comment (riastrad) [reply index.js]: It absolutely does. Good catch, @springmeyer! 🙇

**Comment by riastrad** [comment]
This only fails Travis because we're still testing support for Node v4. All other versions are passing - so I'm going to go ahead an roll out the dev release.

**Comment by riastrad** [comment]
This PR has run it's course.

**Pull request closed without merging**